p2p/pex/reactor.go

```diff
@@ -76,31 +129,149 @@ func (r *ReactorV2) OnStop() {
 	<-r.peerUpdates.Done()
 }
 
+// NextRequestTime returns the next time the pex reactor expects to send a
+// request. Used mainly for testing
+func (r *ReactorV2) NextRequestTime() time.Time {
+	return r.nextRequestTime
+}
+
+// processPexCh implements a blocking event loop where we listen for p2p
+// Envelope messages from the pexCh.
+func (r *ReactorV2) processPexCh() {
```

Comment:
This method is a bit different from how the other reactors are built. Other reactors have separate methods/threads for processing messages and peer updates. Was this done in purpose here?

Reply:
Yeah initially it was like this but I decided to merge them so the reactor was single threaded and thus I don't have to worry about adding mutexes to some of the data structs.

I'm totally fine with splitting them back as two separate threads if we want to. I just try to avoid mutexes if possible

Reply:
@alexanderbez what's your view here. Should I split this into two threads or keep it single threaded?

Reply:
I would vote for definitely splitting it and being consistent with the other reactors.

Reply:
ok. I will make the change